src/renderer/components/settings-execution.tsx

```diff
@@ -12,29 +14,69 @@ import * as React from 'react';
 import { IPackageManager } from '../npm';
 import { AppState } from '../state';
 
+export enum SettingItemType {
+  EnvVar,
+  Flag,
+}
+
 interface ExecutionSettingsProps {
   appState: AppState;
 }
 
+interface ExecutionSettingsState {
+  executionFlags: Record<string, string>;
+  environmentVariables: Record<string, string>;
+}
+
 /**
  * Settings content to manage execution-related preferences.
  *
  * @class ExecutionSettings
  * @extends {React.Component<ExecutionSettingsProps, {}>}
  */
 @observer
-export class ExecutionSettings extends React.Component<ExecutionSettingsProps> {
+export class ExecutionSettings extends React.Component<
+  ExecutionSettingsProps,
+  ExecutionSettingsState
+> {
   constructor(props: ExecutionSettingsProps) {
     super(props);
 
+    this.state = {
+      executionFlags: Object.assign(
+        {},
+        ...props.appState.executionFlags.map((flag, idx) => {
+          return { [`flag-${idx}`]: flag };
+        }),
+      ),
+      environmentVariables: Object.assign(
+        {},
+        ...props.appState.environmentVariables.map((envVar, idx) => {
+          return { [`var-${idx}`]: envVar };
```

Comment:
Converting an array to an object with keys of `var-${idx}` is unintuitive. Is the object form necessary, or could we skip the intermediary and use the arrays here instead?

Reply:
ah yeah i could just do `{ [idx]: envVar }` - the limitation though is that we need to know when we're _updating_ a flag/var at a specific index and not adding a new flag/var; there might be a better way to do that but that's what motivated this

Reply:
cleaned this up a bit :)